libr/core/cmd_search.c

```diff
@@ -1764,7 +1784,8 @@ static void do_syscall_search(RCore *core, struct search_parameters *param) {
 			esp32 = r_str_newf ("%s,=", reg);
 		}
 	}
-	r_list_foreach (list, iter, map) {
+
+	r_list_foreach (param->boundaries, iter, map) {
```

Comment:
i think this is the only change from your PR that must be merged. the other stuff is fixed in a better way in my pr

Reply:
Ill add this change to this other PR https://github.com/radare/radare2/pull/11103 and close this pr